test/integration/scheduler/preemption_test.go

```diff
@@ -181,6 +193,147 @@ var _ = ginkgo.Describe("Preemption", func() {
+				util.FinishEvictionForWorkloads(ctx, k8sClient, lowWl)
+				util.SetWorkloadsAdmissionCkeck(ctx, k8sClient, highWl, checkAnytime.Name, metav1.ConditionTrue, kueue.CheckStateReady)
+				util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, cq.Name, highWl)
+			})
+		})
+
+		ginkgo.It("Should not preempt immediately when on-demand check is used, after succeed", func() {
+			ginkgo.By("Adding the check to the ClusterQueue", func() {
+				cqKey := client.ObjectKeyFromObject(cq)
+				gomega.Eventually(func() error {
+					updatedQueue := &kueue.ClusterQueue{}
+					err := k8sClient.Get(ctx, cqKey, updatedQueue)
+					if err != nil {
+						return err
+					}
+					updatedQueue.Spec.AdmissionChecks = []string{checkOnDemand.Name}
+					return k8sClient.Update(ctx, updatedQueue)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			lowWl := testing.MakeWorkload("low-wl", ns.Name).
+				Queue(q.Name).
+				Priority(lowPriority).
+				Request(corev1.ResourceCPU, "4").
+				Obj()
+
+			ginkgo.By("Creating the first workload wait its admission", func() {
+				gomega.Expect(k8sClient.Create(ctx, lowWl)).To(gomega.Succeed())
+				util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, cq.Name, lowWl)
+				util.SetWorkloadsAdmissionCkeck(ctx, k8sClient, lowWl, checkOnDemand.Name, metav1.ConditionTrue, kueue.CheckStateReady)
+				util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, cq.Name, lowWl)
+			})
+
+			highWl := testing.MakeWorkload("high-wl", ns.Name).
+				Queue(q.Name).
+				Priority(highPriority).
+				Request(corev1.ResourceCPU, "4").
+				Obj()
+			ginkgo.By("Creating a high priority Workload", func() {
+				gomega.Expect(k8sClient.Create(ctx, highWl)).To(gomega.Succeed())
+				util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, cq.Name, highWl)
+				gomega.Consistently(func() bool {
+					readWl := &kueue.Workload{}
+					err := k8sClient.Get(ctx, client.ObjectKeyFromObject(lowWl), readWl)
+					if err != nil {
+						return false
+					}
+					return apimeta.IsStatusConditionTrue(readWl.Status.Conditions, kueue.WorkloadAdmitted)
+
+				}, util.ConsistentDuration, util.Interval).Should(gomega.BeTrue())
+				util.SetWorkloadsAdmissionCkeck(ctx, k8sClient, highWl, checkOnDemand.Name, metav1.ConditionTrue, kueue.CheckStateReady)
+				util.FinishEvictionForWorkloads(ctx, k8sClient, lowWl)
+				util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, cq.Name, highWl)
+			})
+		})
+
+		ginkgo.It("Should not preempt immediately when on-demand check is used, after demand", func() {
+			ginkgo.By("Adding the check to the ClusterQueue", func() {
+				cqKey := client.ObjectKeyFromObject(cq)
+				gomega.Eventually(func() error {
+					updatedQueue := &kueue.ClusterQueue{}
+					err := k8sClient.Get(ctx, cqKey, updatedQueue)
+					if err != nil {
+						return err
+					}
+					updatedQueue.Spec.AdmissionChecks = []string{checkOnDemand.Name}
+					return k8sClient.Update(ctx, updatedQueue)
```

Comment:
It seems the test modified the queue shared with other tests. This can be tricky to debug or run them in parallel. Can we use a dedicated queue for this?

Reply:
cq is created (both as go object and as kubernetes object) before each test and deleted after. This is the way most of the integrations tests are done. 

This should be save according to [ginkgo doc](https://onsi.github.io/ginkgo/#spec-parallelization)

Reply:
yes, this is all good then